ccm_web/client/src/components/RoleSelect.tsx

```diff
@@ -0,0 +1,51 @@
+import React from 'react'
+import { Grid, MenuItem, makeStyles, Select, Typography } from '@material-ui/core'
+
+import { ClientEnrollmentType } from '../models/canvas'
+
+const useStyles = makeStyles((theme) => ({
+  spacing: {
+    marginBottom: theme.spacing(2)
+  }
+}))
+
+interface RoleSelectProps {
+  disabled: boolean | undefined
+  posRoles: ClientEnrollmentType[]
+  selectedRole: ClientEnrollmentType | undefined
+  onRoleChange: (role: ClientEnrollmentType) => void
+}
+
+export default function RoleSelect (props: RoleSelectProps): JSX.Element {
+  const classes = useStyles()
+  const handleRoleSelectChange = (event: React.ChangeEvent<{ value: unknown }>): void => {
+    const value = event.target.value
+    if (typeof value === 'string' && Object.values<string>(ClientEnrollmentType).includes(value)) {
+      props.onRoleChange(value as ClientEnrollmentType)
+    }
+  }
+
+  return (
+    <>
+    <Typography id='role-select-label' className={classes.spacing}>
+      Select a Canvas role that the user will have in the section.
+      Only roles less privileged than your own are allowed.
+    </Typography>
```

Comment:
How do we plan to document for user on which role/enrollment types they can use? I've been using "Only roles less privileged than your own are allowed", but that actually isn't accurate, since instructors I guess can add other instructors. I think we could say "See the docs for info on which roles you can use". I guess this is more important for the bulk process, since we will need to throw a validation error if they provide one they can't use. Here, we'll just show them what they can, although they may have questions about their options.

Reply:
@pushyamig, what's our plan here?

Reply:
I don't think the second line is needed and is communicating wrong info too.  I think for single add user case the dropdown is clear indicator what role he can add. 
 I see the CSV feature is validating if user is trying to add a role if he is not supposed to. We might provide this info in documentation

I won't hold this issue back just for deciding on this.

Reply:
I've already updated the text to be more accurate:
```
    <Typography id='role-select-label' className={classes.spacing}>
      Select a Canvas role that the user will have in the section.
      The available roles are determined by your own role(s) in the course.
    </Typography>
```